Sources/MapboxMaps/Location/Puck/Puck3D.swift

```diff
@@ -89,14 +94,72 @@ internal final class Puck3D: Puck {
             try! style.addSource(source, id: Self.sourceID)
         }
 
+        // Mercator scale
+        let needsUpdateModelScale = updateMercatorScaleIfNeeded(at: location.coordinate.latitude)
+
         // create the layer if needed
         if !style.layerExists(withId: Self.layerID) {
             var modelLayer = ModelLayer(id: Self.layerID)
             modelLayer.source = Self.sourceID
+            modelLayer.modelScale = modelScale
             modelLayer.modelType = .constant(.locationIndicator)
-            modelLayer.modelScale = configuration.modelScale
             modelLayer.modelRotation = configuration.modelRotation
             try! style.addPersistentLayer(modelLayer, layerPosition: nil)
+        } else if needsUpdateModelScale {
+            try? style.setLayerProperty(
+                for: Self.layerID,
+                property: "model-scale",
+                value: modelScale?.toJSON() as Any)
```

Comment:
probably you should also update the model rotation as well, the model rotation is driven by the location updates as well.

Reply:
Resolved as the bearing is updated through model source property, which is more performant.